src/main/java/appeng/tile/storage/TileChest.java

```diff
@@ -259,37 +263,38 @@ else if( fluidCell != null )
 			}
 		}
 
-		switch( channel )
+		if( channel == AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) )
 		{
-			case FLUIDS:
-				if( this.fluidCell == null )
-				{
-					throw NO_HANDLER;
-				}
-				return this.fluidCell;
-			case ITEMS:
-				if( this.itemCell == null )
-				{
-					throw NO_HANDLER;
-				}
-				return this.itemCell;
-			default:
+			if( this.itemCell == null )
+			{
+				throw NO_HANDLER;
+			}
+			return this.itemCell;
+		}
+		else if( channel == AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) && this.fluidCell != null )
+		{
+			if( this.fluidCell == null )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2583)